### PR TITLE
fix fp8 update_weights for block_quant

### PR DESCRIPTION
**Pull request by GuoweiWangU**
## Motivation
After sglang successfully deploys the fp8 block quant model, an error will be reported when calling the update_weights related interface.
```
File "/workspace/bin/sglang/python/sglang/srt/model_loader/weight_utils.py", line 535, in default_weight_loader
      assert param.size() == loaded_weight.size()
```
## Modifications
Delete re-register-parameters for fp8 block quant linear.

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

**Review by BBuf — APPROVED** [review]
LGTM!

**Comment by tianyuzhou95** [comment]
why not just change the `.data` field, like
```diff
diff --git a/python/sglang/srt/layers/quantization/fp8.py b/python/sglang/srt/layers/quantization/fp8.py
index f2c0d6139..02400b107 100644
--- a/python/sglang/srt/layers/quantization/fp8.py
+++ b/python/sglang/srt/layers/quantization/fp8.py
@@ -340,10 +340,8 @@ class Fp8LinearMethod(LinearMethodBase):
                 return
             else:
                 weight, weight_scale = layer.weight.data, layer.weight_scale_inv.data
-            layer.weight = torch.nn.Parameter(weight, requires_grad=False)
-            layer.weight_scale_inv = torch.nn.Parameter(
-                weight_scale, requires_grad=False
-            )
+            layer.weight.data = weight
+            layer.weight_scale_inv.data = weight_scale
             return
```

**Comment by BBuf** [comment]
> why not just change the `.data` field, like
> 
> ```diff
> diff --git a/python/sglang/srt/layers/quantization/fp8.py b/python/sglang/srt/layers/quantization/fp8.py
> index f2c0d6139..02400b107 100644
> --- a/python/sglang/srt/layers/quantization/fp8.py
> +++ b/python/sglang/srt/layers/quantization/fp8.py
> @@ -340,10 +340,8 @@ class Fp8LinearMethod(LinearMethodBase):
>                  return
>              else:
>                  weight, weight_scale = layer.weight.data, layer.weight_scale_inv.data
> -            layer.weight = torch.nn.Parameter(weight, requires_grad=False)
> -            layer.weight_scale_inv = torch.nn.Parameter(
> -                weight_scale, requires_grad=False
> -            )
> +            layer.weight.data = weight
> +            layer.weight_scale_inv.data = weight_scale
>              return
> ```

I think this modification is indeed better. Could we change it to only modify the .data part to avoid copying meta information? @GuoweiWangU

**Comment by tianyuzhou95** [comment]
So this PR aims to keep things unchanged in the case of ROCm, while in other cases, it avoids modifying layer.weight to support weight updates.

Just curious about this  https://github.com/sgl-project/sglang/pull/8390#discussion_r2233102141 from gemini-code-assist. If it is true, why current CI would pass in this case?
